Source/Core/VideoCommon/AsyncRequests.cpp

```diff
@@ -18,6 +19,10 @@ AsyncRequests::AsyncRequests() : m_enable(false), m_passthrough(true)
 
 void AsyncRequests::PullEventsInternal()
 {
+  // This is only called if the queue isn't empty.
+  // So just flush the pipeline to get accurate results.
+  g_vertex_manager->Flush();
```

Comment:
This should probably be stored to a member variable in the constructor instead of calling off the global directly, so it makes dependencies explicit from the class interface.

Reply:
The AsyncRequests instance is a global singleton as well, so fetching any pointer in the constructor isn't a good way either.

But indeed, we should drop this singleton here as well and store all of the videocommon objects in some central way. But I fear this is out of scope of this PR.